src/mac/process.rs

```diff
@@ -8,6 +8,51 @@ use std::fmt::{self, Formatter, Debug};
 use libc::{c_int, gid_t, kill, pid_t, uid_t};
 
 #[derive(Clone)]
+pub enum ProcessStatus {
```

Comment:
I'm not sure where to place the common code yet.  If you have a suggestion, I wouldn't mind hearing. The last part is setting the status value.